lib/ice_cube/builders/string_builder.rb

```diff
@@ -13,14 +13,18 @@ def piece(type, prefix = nil, suffix = nil)
     end
 
     def to_s
-      @types.each_with_object(@base || '') do |(type, segments), str|
+      string = @base || ''
+      @types.each do |type, segments|
```

Comment:
Hey, I can't figure out why your changes are not producing the same test results ... and have no time to spare on this for a few days. Would you add to the PR yourself?